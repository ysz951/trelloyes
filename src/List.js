import React from 'react';
import Card from './Card'
import './List.css';

export default function List(props) {
  
  const renderCard = props.cards.map((card) =>
    <Card
      key={card.id}
      id={card.id}
      listId = {props.id}
      title={card.title}
      content={card.content}
      onClickDelete={props.onClickDelete}
    />)
 
  return (
    <section className='List'>
      <header className='List-header'>
        <h2>{props.header}</h2>
      </header>
      <div className='List-cards'>
        {renderCard}
        <button
          type='button'
          className='List-add-button'
          onClick={() => props.onClickAdd(props.id)}
        >
          + Add Random Card
        </button>
      </div>
    </section>
  )
}

List.defaultProps = {
  onClickAdd: () => {},
  cards: [],
};


