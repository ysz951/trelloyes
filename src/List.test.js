import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';
import STORE from './store';

describe('List component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<List />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    for (let i = 0; i < 4; i++){
      let tree = renderer
        .create(<List header={STORE.lists[i].header} cards = {STORE.lists[i].cardIds.map((id) => STORE.allCards[id])}/>)
        .toJSON();
      expect(tree).toMatchSnapshot();  
    }
  });

});