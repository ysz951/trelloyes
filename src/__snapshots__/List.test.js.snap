// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      First list
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Second card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Fifth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Sixth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Seventh card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Tenth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Twelfth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Thirteenth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <button
      className="List-add-button"
      onClick={[Function]}
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;

exports[`List component renders the UI as expected 2`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Second list
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Second card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Third card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Fourth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Sixth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Eighth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Ninth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Eleventh card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <button
      className="List-add-button"
      onClick={[Function]}
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;

exports[`List component renders the UI as expected 3`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Third list
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Second card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Third card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Fourth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Fifth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Sixth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Seventh card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Eighth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Ninth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Tenth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Eleventh card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Twelfth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Thirteenth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <button
      className="List-add-button"
      onClick={[Function]}
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;

exports[`List component renders the UI as expected 4`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Fourth list
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Twelfth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        Thirteenth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <button
      className="List-add-button"
      onClick={[Function]}
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
